tf2_eigen/CMakeLists.txt

```diff
@@ -1,29 +1,19 @@
 cmake_minimum_required(VERSION 3.5)
 project(tf2_eigen)
 
-find_package(catkin REQUIRED COMPONENTS
-  cmake_modules
-  geometry_msgs
-  tf2
-)
+find_package(ament_cmake REQUIRED)
 
-find_package(Eigen REQUIRED)
+find_package(geometry_msgs REQUIRED)
+find_package(tf2 REQUIRED)
+
+find_package(Eigen3 REQUIRED)
 include_directories(${EIGEN_INCLUDE_DIRS})
 add_definitions(${EIGEN_DEFINITIONS})
```

Comment:
you can look at what we did for `orocos_kdl`. I am not saying that this is the way to go, but it works:
https://github.com/ros2/orocos_kinematics_dynamics/blob/ros2/orocos_kdl/CMakeLists.txt

The cmake variables for Eigen are just a bit special :)